src/cfnlint/__init__.py

```diff
@@ -591,11 +591,25 @@ def get_valid_getatts(self):
 
         return results
 
+    def _rule_enforcement_filter(self, ignore_rules, enforce_rules):
+        result = []
+        for ignore_rule in ignore_rules:
+            # Checking enforce_rule starts with ignore_rule since enforcement uses explicit rules without prefixes
+            match = [enforce_rule for enforce_rule in enforce_rules if enforce_rule.startswith(ignore_rule)]
+            if len(match) == 0:
+                result.append(ignore_rule)
+        return result
+
     def get_directives(self):
         """ Get Directives"""
         results = {}
         for _, resource_values in self.template.get('Resources', {}).items():
             ignore_rule_ids = resource_values.get('Metadata', {}).get('cfn-lint', {}).get('config', {}).get('ignore_checks', [])
+            try:
+                mandatory_rule_ids = cfnlint.config.ConfigMixIn.mandatory_checks
```

Comment:
@kddejong Do you know what the correct way to do this in this scope would be? Since doing it like this returns the object reference since it's an un-initialized class.